python/src/ir.cc

```diff
@@ -1545,6 +1545,15 @@ void init_triton_ir(py::module &&m) {
       .def("run", [](mlir::PassManager &self, mlir::ModuleOp &mod) {
         // TODO: maybe dump module to file and print error for better
         // diagnostics
+        if (auto reproducerPath =
+            ::triton::tools::getenv("TRITON_REPRODUCER_PATH");
+            !reproducerPath.empty()) {
```

Comment:
nit: the syntax is a bit weird, I would move `auto reproducerPath =           ::triton::tools::getenv("TRITON_REPRODUCER_PATH");` above the `if`